### ISSUE TITLE
Client status monitoring

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal#263]
While general monitoring is a broader topic (see https://github.com/fractal-analytics-platform/fractal/issues/224), is there a simple way we could add some status monitoring to the client? Such that a user could request updates after submitting a job and know things like:
1. Is my job still running? / Have there been any errors?
2. How long has my job been running?
3. What task is currently being run?

**Comment by jluethi** [comment]
And in the same logic: If there were errors, can the user check those error messages using the client?

I'm just imagining the demo scenario. If 10 different users submit something to the server and set some parameters wrong, it's impossible from the server log to know which error was affecting which user

**Comment by tcompa** [comment]
The current status of logging in `fractal-server` is the one described in https://github.com/fractal-analytics-platform/fractal-server/issues/52#issuecomment-1268166445, that is:

> * What did fractal-server do for this workflow (e.g. it defined a set of apps, it started running, it completed running). This is currently in server-dir/logs/workflow_XXXXXX/workflow.log.
> * What did each task do (tasks may include some internal logging (e.g. "I created /some/path/plate.zarr"), which should be accessible. This is currently distributed in the files in server-dir/logs/workflow_XXXXXX/scripts.
> * Task errors (e.g. a task raises a ValueError). This currently goes only to the server log stream, which is not useful to retrieve it on a per-workflow level. For testing, we can always use the dummy_fail task, which simply raises an exception with a message.
> * SLURM errors (e.g. memory error). This currently goes in some of the files in the files in server-dir/logs/workflow_XXXXXX/scripts. For testing, we can submit a labeling task to a cpu-low executor, and we'll get something like "Virtual Memory Error".
> * Monitoring of resources (especially memory), either in real-time or ex-post. This is written in the monitoring.db parsl database, but at the moment we are not able to retrieve it in a simple way.

Let's keep the current issue open to decide what the client should do within this scheme. Should there be some `check_info` client command that points to some of the content mentioned above?

**Comment by jluethi** [comment]
Having a client command to check this would be great! Something like `check_status` and then a way to get the relevant logs :)

**Comment by tcompa** [comment]
With the current logs, the easiest way to serve logs would be to merge all the `stderr` files in a specific folder and then sort their lines by time. This would offer a comprehensive log stream, without thinking too much about how files are organized (which is related to parsl executor blocks, and at the moment is not totally transparent).

As for the status, this is probably better discussed after https://github.com/fractal-analytics-platform/fractal-server/issues/107.

**Comment by jluethi** [comment]
That's certainly a good start. A user would then request the log for a given workflow, which makes sense :)

**Comment by tcompa** [comment]
> Once [fractal-analytics-platform/fractal-server#327](https://github.com/fractal-analytics-platform/fractal-server/pull/327) is merged, we should add the corresponding client command that queries the status of a worfklow-execution job

Note that this will offer "fractal" information, and not "slurm" information. This means that a status "submitted" (or "running") correspond to "fractal prepared this workflow and gave it to a runner backend to run". Having a detailed view on the slurm statuses is a different feature, which we may consider somewhere else. Meanwhile, there are other options to get a granular view on what task is running (ref https://github.com/fractal-analytics-platform/fractal-server/issues/328), without using `squeue` or similar.

Broadly speaking, this means that we try to keep the runner backend as an isolated component, and we'd rather not include any slurm-specific statement outside the corresponding backend.

**Comment by tcompa** [comment]
This is currently complete, through the `fractal workflow job-status` command.
Re-opening until we release this new version.

**Comment by tcompa** [comment]
Now available in `1.0.0a5`, to be used as
```
fractal workflow job-status --job-id $JOB_ID
```
or
```
fractal --batch workflow job-status --job-id $JOB_ID
```

(where `JOB_ID` can be read from the output of `fractal workflow apply`)
